lang/cs/Org.Apache.REEF.IMRU/API/IIMRUResultHandler.cs

```diff
@@ -0,0 +1,39 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using Org.Apache.REEF.IMRU.OnREEF.ResultHandler;
+using Org.Apache.REEF.Network.Group.Task.Impl;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IMRU.API
+{
+    /// <summary>
+    /// Interface defining how to handle the result
+    /// </summary>
+    /// <typeparam name="TResult">Generic type</typeparam>
+    public interface IIMRUResultHandler<in TResult>
```

Comment:
@markusweimer I just realized a problem with having IObserver<int>. Consider MapperCount example, I believe in this case, IObserver<int> is already bounded to something and hence I will get duplicate key exception from Tang while instantiating it.